basyx/aas/adapter/http.py

```diff
@@ -600,42 +615,32 @@ def _get_shells(self, request: Request) -> Iterator[model.AssetAdministrationShe
             aas = filter(lambda shell: all(specific_asset_id in shell.asset_information.specific_asset_id
                          for specific_asset_id in specific_asset_ids), aas)
 
-        return aas
+        paginated_aas, end_index = self._get_slice(request, aas)
+        return paginated_aas, end_index
 
     def _get_shell(self, url_args: Dict) -> model.AssetAdministrationShell:
         return self._get_obj_ts(url_args["aas_id"], model.AssetAdministrationShell)
 
     def _get_submodels(self, request: Request) -> Tuple[Iterator[model.Submodel], int]:
-        limit = request.args.get('limit', type=int, default=10)
-        cursor = request.args.get('cursor', type=int, default=0)
-        submodels: List[model.Submodel] = list(self._get_all_obj_of_type(model.Submodel))
-        # Apply pagination
-        start_index = cursor
-        end_index = cursor + limit
-        paginated_submodels: Iterator[model.Submodel] = iter(submodels[start_index:end_index])
+        submodels: Iterator[model.Submodel] = self._get_all_obj_of_type(model.Submodel)
         id_short = request.args.get("idShort")
         if id_short is not None:
-            paginated_submodels = filter(lambda sm: sm.id_short == id_short, paginated_submodels)
+            submodels = filter(lambda sm: sm.id_short == id_short, submodels)
         semantic_id = request.args.get("semanticId")
         if semantic_id is not None:
             spec_semantic_id = HTTPApiDecoder.base64urljson(semantic_id, model.Reference, False)  # type: ignore
-            paginated_submodels = filter(lambda sm: sm.semantic_id == spec_semantic_id, paginated_submodels)
-        return (paginated_submodels, end_index)
+            submodels = filter(lambda sm: sm.semantic_id == spec_semantic_id, submodels)
+        paginated_submodels, end_index = self._get_slice(request, submodels)
+        return paginated_submodels, end_index
 
     def _get_submodel(self, url_args: Dict) -> model.Submodel:
         return self._get_obj_ts(url_args["submodel_id"], model.Submodel)
 
-    def _get_submodel_submodel_elements(self,  request: Request, url_args: Dict) ->\
+    def _get_submodel_submodel_elements(self, request: Request, url_args: Dict) ->\
             Tuple[List[model.SubmodelElement], int]:
-        limit = request.args.get('limit', type=int, default=10)
-        cursor = request.args.get('cursor', type=int, default=0)
         submodel = self._get_submodel(url_args)
-        submodelelements = list(submodel.submodel_element)
-        # Apply pagination
-        start_index = cursor
-        end_index = cursor + limit
-        paginated_submodelelements = submodelelements[start_index:end_index]
-        return (paginated_submodelelements, end_index)
+        paginated_submodelelements, end_index = self._get_slice(request, submodel.submodel_element)
+        return list(paginated_submodelelements), end_index
```

Comment:
`_get_submodels` and `_get_shells` both return iterators, but this function returns a list instead. Can you make this consistent? I'm okay with either variant.